### ISSUE TITLE
Vmagent performance degradation after restarting

### ISSUE THREAD

**Issue by laixintao** [VictoriaMetrics/VictoriaMetrics#6644]
### Describe the bug

I have 12 vmagents collecting metrics from a group of targets.

Today I upgrade one of them from v1.93.9 to v1.101.0, the CPU usage of that vmagent increased a lot.

<img width="1914" alt="image" src="https://github.com/user-attachments/assets/d83cdb18-7c59-44e2-b9ac-38d177ad7e58">

### To Reproduce

start command of vmagent:

```shell
/opt/xx-vmagent/vmagent-prod \
  -httpListenAddr=":5429" \
  -promscrape.config=/etc/xx-vmagent/prom-conf.yaml \
  -promscrape.cluster.membersCount=12 \
  -promscrape.cluster.memberNum=0 \
  -promscrape.maxScrapeSize=50MiB \
  -remoteWrite.label='shard=0' \
  -remoteWrite.url="http://127.0.0.1:5480/insert/0/prometheus/api/v1/write" \
  -promscrape.httpSDCheckInterval=5m \
  -promscrape.streamParse=false \
  -remoteWrite.rateLimit=50000000 \
  -remoteWrite.streamAggr.keepInput \
  -remoteWrite.streamAggr.config /etc/xx-vmagent/stream-aggr-conf.yaml \
  -remoteWrite.tmpDataPath="/home/vmagent_xx_cluster_0"
```

and my stream config:

```shell
cat /etc/xx-vmagent/stream-aggr-conf.yaml
-   by:
    - instance
    interval: 30s
    match: node_cpu_seconds_total{job="node", tag_application!="", mode!="idle"}
    output_relabel_configs:
    -   replacement: '0'
        target_label: shard
    outputs:
    - total
    staleness_interval: 10m
-   by:
    - instance
    interval: 30s
    match: node_cpu_seconds_total{tag_application!="", job="node", mode="idle"}
    output_relabel_configs:
    -   replacement: '0'
        target_label: shard
    outputs:
    - count_series
    staleness_interval: 10m

```

### Version

old (better performance)

vmagent-20231210-124303-tags-v1.93.9-0-gb0de9c874

now (poor performance)

vmagent-20240425-145459-tags-v1.101.0-0-g5334f0c2c

### Additional information

pprof flamegraph shows that the `scrapeInterval` was difference than before:

Old 1.93 one

<img width="2517" alt="image" src="https://github.com/user-attachments/assets/3f6791d5-0cfc-4171-8394-3c30231a95ef">

Now 1.101 one

<img width="2511" alt="image" src="https://github.com/user-attachments/assets/380d4cd5-f069-4613-a86c-d32978fe157e">

pprof files:

[profile-of-two-vmagent-versions.zip](https://github.com/user-attachments/files/16232618/profile-of-two-vmagent-versions.zip)

**Comment by laixintao** [comment]
Turns out it's not the version issue, I restart one 1.93 vmagent, after restarting, the CPU increased a lot as well.

But I still can figure it out why restarting would cause CPU changes, (any JIT magic in vmagent?)

**Comment by Amper** [comment]
Hey, @laixintao.
Is the increased CPU consumption temporary after startup?
If so, it is probably related to the Service Discovery work.

**Comment by laixintao** [comment]
> Hey, @laixintao. Is the increased CPU consumption temporary after startup? If so, it is likely related to the Service Discovery work.

yeah I thought about this as well. But sadly no, it stays high cpu usage till now.

My dashboard for vmagent:

<img width="980" alt="image" src="https://github.com/user-attachments/assets/74bad094-1523-480a-8bea-f40a33598ca0">

The cpu usage stays high

<img width="1736" alt="image" src="https://github.com/user-attachments/assets/5536e766-262b-4ed1-b456-1219a30f0730">

(oh just found that one (green line) seems reduced and became unstable, going to try to find out why)

**Comment by valyala** [comment]
@laixintao , try upgrading vmagent to [v1.102.0](https://docs.victoriametrics.com/changelog/), and reporting CPU usage with this release. The 1.102.0 release contains some performance improvements and resource usage improvements in both [streaming aggregation](https://docs.victoriametrics.com/stream-aggregation/) and Prometheus targets' scraping.

**Comment by laixintao** [comment]
> @laixintao , try upgrading vmagent to [v1.102.0](https://docs.victoriametrics.com/changelog/), and reporting CPU usage with this release. The 1.102.0 release contains some performance improvements and resource usage improvements in both [streaming aggregation](https://docs.victoriametrics.com/stream-aggregation/) and Prometheus targets' scraping.

Nice, thanks, will do

**Comment by laixintao** [comment]
> @laixintao , try upgrading vmagent to [v1.102.0](https://docs.victoriametrics.com/changelog/), and reporting CPU usage with this release. The 1.102.0 release contains some performance improvements and resource usage improvements in both [streaming aggregation](https://docs.victoriametrics.com/stream-aggregation/) and Prometheus targets' scraping.

The improvement was remarkable! thanks.

<img width="2193" alt="image" src="https://github.com/user-attachments/assets/a25b4426-65f4-4d2e-bdc8-3d837847e497">

I am closing this issue, thank you so much.

One more strange thing, actually before 9 July all my servers was running on 70%-80% CPU usages, each one collecting thousands of targets, without any changes (at least as I remembered), the CPU usage was dropped to 30%.

<img width="2190" alt="image" src="https://github.com/user-attachments/assets/1a53706f-7b8e-4fc7-bd46-49e5844a302a">

Till today, the CPU stays around 30%, but once I restart the vmagent process, CPU usage goes to 80%. Don't know what's happening.

- Checked the htop on server, it's indeed 80%, so not a monitoring issue
- Also check the 30% ones, htop shows 30%, so not a monitoring issue
- Checked metrics of it was collecting, correct
- All 30% CPU ones and 80% ones are collecting the same metrics (this is the issue I described here)

Really weird.

But I decide I don't care anymore, with 1.102 huge improvement, I am going to upgrade all of my vmagents.
